### PR TITLE
chore(devimint): faster and cleaner start

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: devimint/src/util.rs [hunk 1]
```diff
@@ -331,25 +332,37 @@ const DEFAULT_RETRIES: usize = 30;
 pub async fn poll<Fut, R>(
     name: &str,
     // TODO: this should be `Duration`, not number of retries --dpc
```
Comment (bradleystachurski) [inline devimint/src/util.rs:334]: You did it! :smile:

File: devimint/src/util.rs [hunk 2]
```diff
@@ -331,25 +332,37 @@ const DEFAULT_RETRIES: usize = 30;
...
-                debug!("polling {name} failed with: {err:?}, will retry... ({i}/{retries})");
-                task::sleep(Duration::from_secs(1)).await;
+            Err(ControlFlow::Continue(err))
+                if attempt_start
+                    .duration_since(start)
+                    .expect("time goes forward")
+                    < timeout =>
+            {
+                debug!(target: LOG_DEVIMINT, %attempt, %err, "Polling {name} failed, will retry...");
+                task::sleep(Duration::from_millis(attempt.min(1000))).await;
```
Comment (bradleystachurski) [inline devimint/src/util.rs:357]: `attempt.min(1000)` is pretty aggressive. Should we multiply attempt by a factor (100?) or did you want to sleep only a few millis?

**Review by bradleystachurski — APPROVED** [review]

File: devimint/src/util.rs (hunk 2 above)
Comment (dpc) [reply devimint/src/util.rs:357]: 1s is not agressive, IMO. All these commands are supposed to wrap in a handful of seconds if everything is OK.

Comment (bradleystachurski) [reply devimint/src/util.rs:357]: Perhaps I'm misreading, but `Duration::from_millis(attempt.min(1000))` will wait 1 milli on the first attempt, then 2 until 1000 attempts.

Comment (dpc) [reply devimint/src/util.rs:357]: Exactly. :D

1 ms + time of running the command

**Review by justinmoon — APPROVED** [review]

**Pull request merged**